website/src/theme/DocItem/index.js

```diff
@@ -0,0 +1,27 @@
+import sha1 from "sha1";
+import React from "react";
+import OriginalDocItem from "@theme-original/DocItem";
+import useDocusaurusContext from "@docusaurus/useDocusaurusContext";
+import { OpenGraphPreview } from "@site/src/shared/lib/open-graph-preview";
+
+function DocItem(props) {
+    const { content } = props;
+    const { id } = content.metadata;
+    const { siteConfig, i18n } = useDocusaurusContext();
+
+    const hashFileName = sha1(id + i18n.currentLocale);
+
+    return (
+        <>
+            <OpenGraphPreview
+                imgUrl={`${siteConfig.url}${
+                    /* OG Preview images build with locale prefix (.../en/assets/...) for not default locales */
+                    i18n.currentLocale !== i18n.defaultLocale ? `/${i18n.currentLocale}` : ""
+                }/assets/og/${hashFileName}.jpg`}
+            />
```

Comment:
[`nitpick(debug):`](https://conventionalcomments.org/) Ну и тут тоже бы подвязаться на флаг, чтобы не перезатирало лишний раз

[`thoughts:`](https://conventionalcomments.org/) + Можно прям по-хорошему это прям в `shared/config` объявить, как и другие переменные окружения

А оттуда импортить потом (чтоб по канону :D )

```suggestion
            {process.env.OG_EXP && <OpenGraphPreview meta={content.metadata} />
```